corehq/motech/openmrs/repeater_helpers.py

```diff
@@ -166,7 +166,40 @@ def get_patient_matching_identifiers(self, patient_identifier_type, patient_iden
 
 
 CaseTriggerInfo = namedtuple('CaseTriggerInfo',
-                             ['case_id', 'updates', 'created', 'closed', 'extra_fields'])
+                             ['case_id', 'updates', 'created', 'closed', 'extra_fields', 'form_question_values'])
+
+
+def get_form_question_values(form_json):
+    """
+    Returns question-value pairs to result where questions are given as "/data/foo/bar"
+    
+    >>> get_form_question_values({'form': {'foo': {'bar': 'baz'}}})
+    {'/data/foo/bar': 'baz'}
+
+    """
+    _reserved_keys = ('@uiVersion', '@xmlns', '@name', '#type', 'case', 'meta', '@version')
+
+    def _recurse_form_questions(form_dict, path, result_):
+        for key, value in form_dict.items():
+            if key in _reserved_keys:
+                continue
+            new_path = path + [key]
+            if isinstance(value, list):
+                # Repeat group
+                for v in value:
+                    assert isinstance(v, dict)
+                    _recurse_form_questions(v, new_path, result_)
+            elif isinstance(value, dict):
+                # Group
+                _recurse_form_questions(value, new_path, result_)
+            else:
+                # key is a question and value is its answer
+                question = '/'.join(new_path)
+                result_[question] = value
+
+    result = {}
+    _recurse_form_questions(form_json['form'], ['/data'], result)
```

Comment:
`'data'` might technically come from `form_json['form']['type']` or something, but form builder always calls it data. I think fine to leave, maybe worth leaving a comment, though even that may be overkill. Maybe no change required, that just came to mind.

Reply:
Good call. Comment added.